pallets/parachain-staking/src/lib.rs

```diff
@@ -1643,21 +1641,17 @@ pub mod pallet {
 		pub(crate) fn balance_frozen_extended(
 			account: &T::AccountId,
 			is_collator: bool,
-		) -> BalanceOf<T> {
-			use frame_support::traits::fungible::InspectFreeze;
-
+		) -> Option<BalanceOf<T>> {
 			// First check and migrate any existing lock
 			// We ignore the result as we want to return the frozen balance regardless
 			let _ = Self::check_and_migrate_lock(account, is_collator);
 
 			// Now return the frozen balance
-			let freeze_reason = if is_collator {
-				FreezeReason::StakingCollator
+			if is_collator {
+				<CandidateInfo<T>>::get(account).map(|info| info.bond)
 			} else {
-				FreezeReason::StakingDelegator
-			};
-
-			T::Currency::balance_frozen(&freeze_reason.into(), account)
```

Comment:
Now, this function does not return the frozen balance

Reply:
Amended https://github.com/moonbeam-foundation/moonbeam/pull/3306/commits/b8aae6ebb8086b81546764f1278734769f74d45c